awslabs/smithy-kotlin
# chore(rt): Update unit test lib dependencies

Use JUnit5 (jupiter) as unit test framework. Clean up test dependences in client-runtime.

*Issue #, if available:* N/A

## Description of changes:* Update MPP test dependencies for JVM to use JUnit5/Jupiter over JUnit 4.

## Testing done:
* Run all tests via gradle in `smithy-kotlin`
* Run all tests via gradle in `aws-sdk-kotin` with composite build enabled

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
